cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionGoal.java

```diff
@@ -73,18 +74,33 @@ int numInterestedReplicas(ClusterModel clusterModel) {
     return _balancingConstraint.replicaBalancePercentage();
   }
 
+  /**
+   * Get the number of brokers that can be dropped if the given cluster is over-provisioned, {@code null} otherwise.
+   *
+   * @param clusterModel The state of the cluster.
+   * @return Number of brokers that can be dropped if the given cluster is over-provisioned, {@code null} otherwise.
+   */
+  private Integer numBrokersToDrop(ClusterModel clusterModel) {
```

Comment:
Shouldn't this just return int?

Reply:
No, JavaDoc indicates why it may return `null`:

```
Get the number of brokers that can be dropped if the given cluster is over-provisioned, {@code null} otherwise.
```

So if the cluster is not over-provisioned, it returns null.